### PR TITLE
[new] Show and hide tabs and controls with control groups

### LINKED ISSUE #61: Add show/hide settings/tabs option to WSF
See how this is handled directly in Flux: https://github.com/iconicwp/flux-checkout/blob/feature/81-replace-material-design-template/source/admin/js/dynamic-settings.js

This uses classes on the fields.

We should be able to add options to the settings and loop over the settings and add those show/hide elements for each group chosen.

### PR DESCRIPTION
**Pull request by mattwatsoncodes**
Fixes #61.

**To show and hide tabs:**

```
$settings['tabs'][] = array(
    'id'              => 'styles',
    'title'           => __( 'Styles', 'flux-checkout' ),
    'tabControlGroup' => 'desktop_enabled',
    // 'showIfValue'     => array( true ), // show if will hide the tab unless the value matches.
    'hideIfValue'     => array( true ), // hide will show the tab unless the value matches.
);
```

On your control, be sure to set the same control name:

```
array(
        'id'              => 'desktop_enabled',
        'title'           => __( 'Enable on Desktop', 'flux-checkout' ),
        'subtitle'        => __( 'Enable or disable Flux Checkout on Desktop.', 'flux-checkout' ),
        'type'            => 'toggle',
        'default'         => true,
        'tabControlGroup' => 'desktop_enabled', // Needs to be set to the same tab control. Does not have to match the Id of this control. 
),
```

**To show and hide controls**

```
array(
        'id'               => 'logo_image',
        'title'            => __( 'Header Image', 'flux-checkout' ),
        'subtitle'         => __( 'Select your logo (Recommended size: 200 x 40px).', 'flux-checkout' ),
        'type'             => 'file',
        'showControlGroup' => 'header',
        'showIfValue'      => array( 'image' ),  // show if will hide the control unless the value matches.
        // 'hideIfValue'      => array( 'text' ),  // hide if will show the control unless the value matches.
),
```

You need to set the control that the values are taken from like this:

```
array(
        'id'               => 'branding',
        'title'            => __( 'Header Type', 'flux-checkout' ),
        'subtitle'         => __( 'Select whether to use an image or text for the branding.', 'flux-checkout' ),
        'type'             => 'select',
        'choices'          => array(
	        'text'  => __( 'Text', 'flux-checkout' ),
	        'image' => __( 'Image', 'flux-checkout' ),
        ),
        'default'          => 'text',
        'class'            => 'header-type',
        'showControlGroup' => 'header', // Needs to be set to the same control. Does not have to match the Id of this control. 
),
```

Demo here: https://cln.sh/Wiwuhw

### CODE REVIEW

**Review by jamesckemp — COMMENTED** [review]
I only have 2 concerns:

1. We use underscores, rather than camelcase, for consitency.
2. How does `showControlGroup` know it's the field to use as a value, rather than just another control to show/hide?

**Comment by mattwatsoncodes** [comment]
@jamesckemp no worries I can solve the case issue fairly quickly. 

With regards to `showControlGroup` knowing it's the field to use as a value, if it does not have `showIfValue` or `hideIfValue` set then it is configured as the parent control.

To be honest, I should probably rename that one to `show_controller` or 'show_control_parent` or something similar.

File: settings/example-settings.php [hunk 1]
```diff
@@ -179,17 +181,72 @@ function wpsf_tabless_settings( $wpsf_settings ) {
 				'default' => 'This is default',
 			),
 		),
+		array(
```
Comment (jamesckemp) [inline settings/example-settings.php:184]: Should this be inside the fields array?

**Review by jamesckemp — CHANGES_REQUESTED** [review]
Looking much better, thanks.

We can hide tabs and fields, but can we hide sections as well?

**Comment by mattwatsoncodes** [comment]
CleanShot 2021-10-06 at 10 54 46

Now with the ability to show and hide sections 👍 @jamesckemp

File: settings/example-settings.php [hunk 2]
```diff
@@ -206,18 +201,112 @@ function wpsf_tabless_settings( $wpsf_settings ) {
...
+				'show_if_value'      => array( 'option-2', 'option-3' ), // show if will hide the control unless the value matches.
+			),
+			array(
+				'id'                 => 'hide-if-option-1',
+				'title'              => 'Hide if Option 1',
+				'subtitle'           => 'Will hide if Option 1 is set.',
+				'type'               => 'select',
+				'type'               => 'text',
+				'default'            => 'This is default',
+				'show_control_group' => 'control-group', // Needs to be set to the control group.
```
Comment (jamesckemp) [inline settings/example-settings.php:262]: I'm wondering if we should call these `field_control_group` just to match the format of tab and section?

**Review by jamesckemp — CHANGES_REQUESTED** [review]

**Comment by mattwatsoncodes** [comment]
@jamesckemp syntax changed as requested 👍 back to you.

**Review by jamesckemp — APPROVED** [review]

**Pull request merged**